tests/test-location-label.c

```diff
@@ -0,0 +1,77 @@
+/* Copyright 2020 Endless Mobile, Inc. */
+
+/* This file is part of eos-metrics-instrumentation.
+ *
+ * eos-metrics-instrumentation is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * eos-metrics-instrumentation is distributed in the hope that it will be
+ * useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+ * Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with eos-metrics-instrumentation.  If not, see
+ * <http://www.gnu.org/licenses/>.
+ */
+
+#include "eins-location-label.h"
+
+static void
+test_empty_keyfile (void)
+{
+  g_autoptr(GKeyFile) kf = g_key_file_new ();
+  g_autoptr(GVariant) payload = build_location_label_event (kf);
+
+  g_assert_null (payload);
+}
+
+static void
+test_empty_group (void)
+{
+  g_autoptr(GKeyFile) kf = g_key_file_new ();
+  gboolean ret = g_key_file_load_from_data (kf, "[Label]\n", -1, G_KEY_FILE_NONE, NULL);
```

Comment:
nit: would be a bit better to move `#define LOCATION_LABEL_GROUP "Label"` to the header file and use that.

Reply:
I'm not sure this line would be any clearer or more maintainable if it were written as:

```suggestion
  gboolean ret = g_key_file_load_from_data (kf,  "[" LOCATION_LABEL_GROUP "]\n", -1, G_KEY_FILE_NONE, NULL);
```

or as:

```suggestion
  g_autofree gchar *data = g_strdup_printf ("[%s]\n", LOCATION_LABEL_GROUP);
  gboolean ret = g_key_file_load_from_data (kf, "[Label]\n", -1, G_KEY_FILE_NONE, NULL);
```

Reply:
On reflection I actively prefer for the test to inline the constant.

Consider what would happen if the test used `LOCATION_LABEL_GROUP` as you suggest, and then someone changed the definition of that constant. The test would still pass – but the application would no longer do the right thing, since it would no longer read the file as it exists on machines in the wild.

So unless you feel strongly otherwise I'd prefer to keep it inlined.

Reply:
Touché